### ISSUE TITLE
Variable is read but never assigned rule should consider code flow

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#893]
If a variable is read before it is assigned a value and a value is assigned after that, the rule currently doesn't trigger.
It should still trigger if the variable is read before the first assignment.

```
METHOD PUBLIC CHARACTER FooMethod(
):
   DEFINE VARIABLE lcFoo AS CHARACTER NO-UNDO.
   DEFINE VARIABLE lcFoo2 AS CHARACTER NO-UNDO.

   lcFoo2 = lcFoo.
   lcFoo = "abc":U.

   RETURN lcFoo2.
END METHOD.
```

**Comment by gquerret** [comment]
Duplicate of #628
Code flow information is now available in the parse tree, so enhancement can now be implemented.
